tests/MongoDB.Bson.Tests/ObjectModel/ObjectIdTests.cs

```diff
@@ -14,20 +14,76 @@
 */
 
 using System;
+using System.Globalization;
 using System.Linq;
-using MongoDB.Bson;
+using FluentAssertions;
+using MongoDB.Bson.TestHelpers;
 using Xunit;
 
 namespace MongoDB.Bson.Tests
 {
     public class ObjectIdTests
     {
+        [Theory]
+        [InlineData(0xFFFFFE, 0xFFFFFF)]
+        [InlineData(0xFFFFFF, 0)]
+        [InlineData(0x1000000, 1)]
+        [InlineData(0x1FFFFFE, 0xFFFFFF)]
+        [InlineData(0x1FFFFFF, 0)]
+        [InlineData(0x2000000, 1)]
+        public void Ensure_that_counter_is_resetted_after_max_value(int incrementValue, int expectedIncrement)
+        {
+            ObjectIdReflector.__staticIncrement(incrementValue);
+            var oid = ObjectId.GenerateNewId();
+            var resultCounter = oid.Increment;
+            resultCounter.Should().Be(expectedIncrement);
+        }
+
+        [Fact]
+        public void Ensure_that_machine_value_is_not_the_same_if_created_from_different_appdomain()
+        {
+#if NET452
+            var parentOid = ObjectId.GenerateNewId();
+
+            ObjectId childOid = default;
+            AppDomain testDomain = null;
+            try
+            {
+                testDomain = AppDomain.CreateDomain("MyDomain", null, new AppDomainSetup { ApplicationBase = AppDomain.CurrentDomain.SetupInformation.ApplicationBase });
+                var marshalType = typeof(AppDomainCourier);
+                var testProxy = (AppDomainCourier)testDomain.CreateInstanceAndUnwrap(marshalType.Assembly.FullName, marshalType.FullName);
+                childOid = testProxy.ObjectId;
+            }
+            finally
+            {
+                if (testDomain != null)
+                {
+                    AppDomain.Unload(testDomain);
+                }
+            }
+
+            childOid.Machine.Should().NotBe(parentOid.Machine);
+#endif
+        }
+
+        [Theory]
+        [InlineData(0x00000000, "January 1, 1970 00:00:00Z")]
+        [InlineData(0x7FFFFFFF, "January 19, 2038 03:14:07Z")]
+        [InlineData(0x80000000, "January 19, 2038 03:14:08Z")]
+        [InlineData(0xFFFFFFFF, "February 7, 2106 06:28:15Z")]
+        public void Ensure_that_timestamp_is_represented_as_unsigned_int(uint timestamp, string expectedDateString)
+        {
+            var oid = ObjectId.GenerateNewId(timestamp);
+            var expectedDate = DateTime.Parse(expectedDateString, CultureInfo.InvariantCulture, DateTimeStyles.AdjustToUniversal | DateTimeStyles.AssumeUniversal);
+            oid.CreationTime.Should().Be(expectedDate);
+        }
```

Comment:
If you accept my suggestions in `ObjectId.cs` this test would more accurately use the word "interpreted" than "represented".

I suggest the following renamings and refactoring:

    [Theory]
    [InlineData(0x00000000, "1970-01-01T00:00:00Z")]
    [InlineData(0x7FFFFFFF, "2038-01-19T03:14:07Z")]
    [InlineData(0x80000000, "2038-01-19T03:14:08Z")]
    [InlineData(0xFFFFFFFF, "2106-02-07T06:28:15Z")]
    public void Ensure_that_timestamp_is_interpreted_as_unsigned_int(uint timestamp, string expectedDateString)
    {
        var objectId = ObjectId.GenerateNewId((int)timestamp);
        var expectedDate = DateTime.Parse(expectedDateString, CultureInfo.InvariantCulture, DateTimeStyles.AdjustToUniversal | DateTimeStyles.AssumeUniversal);
        objectId.CreationTime.Should().Be(expectedDate);
    }

Note the use of ISO standard date strings and the cast to `int` in the call to `GenerateNewId`.